keras/engine/training.py

```diff
@@ -1855,8 +1857,6 @@ class during training. This can be useful to tell the model to "pay
       self.train_function = self.make_train_function()
       logs = self.train_function(iterator)
 
-    if reset_metrics:
```

Comment:
I think this is probably needed as well, otherwise the follow up model.eval() will accumulate the metric result from this train/test_on_batch.

Reply:
See the extended check, it seems not.

Reply:
ok. Good to know.